### PR TITLE
fix camlp4 build after ocaml/ocaml#1118

### PR DESCRIPTION
**Pull request by gasche**

### CODE REVIEW

File: camlp4/Camlp4/Struct/Camlp4Ast2OCamlAst.ml [hunk 1]
```diff
@@ -790,7 +794,7 @@ value varify_constructors var_names =
       | Ptyp_constr longident lst ->
           Ptyp_constr longident (List.map loop lst)
       | Ptyp_object (lst, o) ->
-          Ptyp_object (List.map (fun (s, a, t) -> (s, a, loop t)) lst, o)
+          Ptyp_object (List.map (fun (Otag s a t) -> (Otag s a (loop t))) lst, o)
```
Comment (objmagic) [inline camlp4/Camlp4/Struct/Camlp4Ast2OCamlAst.ml]: Code here looks like [`Ast_helper.varify_constructors`](https://github.com/ocaml/ocaml/blob/aa6b5b96a5dd41dbace23d696c2670f7a296d112/parsing/ast_helper.ml#L90). Should we match `Oinherit` here as well?

File: camlp4/boot/Camlp4.ml [hunk 2]
```diff
@@ -16269,7 +16270,7 @@ module Struct =
                       Ptyp_constr (longident, (List.map loop lst))
                   | Ptyp_object ((lst, o)) ->
                       Ptyp_object
-                        (((List.map (fun (s, a, t) -> (s, a, (loop t))) lst),
+                        (((List.map (fun (Otag (s, a, t)) -> Otag (s, a, (loop t))) lst),
```
Comment (objmagic) [inline camlp4/boot/Camlp4.ml:16273]: Should we match `Oinherit` here as well?

**Comment by objmagic** [comment]
Actually, I am not familiar with camlp4 source code at all. So it's better to have someone else to review this.

**Comment by gasche** [comment]
@objmagic this PR just fixes the build, and I believe that it passes the "obviously correct" test (the CI failed because I did a small mistake with revised syntax, but I just fixed it). We should indeed try to properly handle in the Camlp4 codebase the new constructs that you introduced, but that is a bit more technical: we need to change the parser to parse the new syntax, adapt the Camlp4 AST format to support it, and then change the Camlp4AST2OCamlAst pass to transform it into your new parsetree construction. We may want to do this, but in a new PR -- and it could make sense to try to share work with [camlp5](https://github.com/camlp5/camlp5).

**Pull request merged**